lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -45,6 +45,13 @@
     description:
       - Description of the security group. Required when C(state) is C(present).
     required: false
+  diff:
```

Comment:
Does this module have its own `diff` output? This seems like a generic return field that is not usually documented per module.

Reply:
I believe it would be categorized as it's own diff output. As when I enable diff: true, I get an output similar to the following:

```
TASK [ec2_securitygroups : Update tomtest SG] ********************************************************************************************************************************************************************************
--- before
+++ after
@@ -4,25 +4,13 @@
     "group_name": "tomtest",
     "ip_permissions": [
         {
-            "from_port": 80,
-            "ip_protocol": "tcp",
-            "ip_ranges": [
-                {
-                    "cidr_ip": "0.0.0.0/0",
-                    "description": "Incoming HTTP/HTTPS"
-                }
-            ],
-            "ipv6_ranges": [],
-            "prefix_list_ids": [],
-            "to_port": 80,
-            "user_id_group_pairs": []
-        },
-        {
             "from_port": 443,
             "ip_protocol": "tcp",
             "ip_ranges": [
                 {
-                    "cidr_ip": "0.0.0.0/0",
+                    "cidr_ip": [
+                        "0.0.0.0/0"
+                    ],
                     "description": "Incoming HTTP/HTTPS"
                 }
             ],
@@ -33,34 +21,6 @@
         }
     ],
     "ip_permissions_egress": [
-        {
-            "from_port": 80,
-            "ip_protocol": "tcp",
-            "ip_ranges": [
-                {
-                    "cidr_ip": "0.0.0.0/0",
-                    "description": "Outgoing to the World"
-                }
-            ],
-            "ipv6_ranges": [],
-            "prefix_list_ids": [],
-            "to_port": 80,
-            "user_id_group_pairs": []
-        },
-        {
-            "from_port": 22,
-            "ip_protocol": "tcp",
-            "ip_ranges": [
-                {
-                    "cidr_ip": "0.0.0.0/0",
-                    "description": "Outgoing to the World"
-                }
-            ],
-            "ipv6_ranges": [],
-            "prefix_list_ids": [],
-            "to_port": 22,
-            "user_id_group_pairs": []
-        },
         {
             "from_port": 25,
             "ip_protocol": "tcp",
@@ -74,20 +34,6 @@
             "prefix_list_ids": [],
             "to_port": 25,
             "user_id_group_pairs": []
-        },
-        {
-            "from_port": 443,
-            "ip_protocol": "tcp",
-            "ip_ranges": [
-                {
-                    "cidr_ip": "0.0.0.0/0",
-                    "description": "Outgoing to the World"
-                }
-            ],
-            "ipv6_ranges": [],
-            "prefix_list_ids": [],
-            "to_port": 443,
-            "user_id_group_pairs": []
         }
     ],
     "owner_id": "#######",

changed: [localhost]```

Reply:
This is normal behavior when `diff` is enabled for a task or via the `--diff` command line parameter. Since this is not part of this module's argument spec, it doesn't belong in the module documentation.